python/ccxt/bybit.py

```diff
@@ -1368,43 +1426,50 @@ def fetch_order(self, id, symbol=None, params={}):
-                else:
-                    amount = cost if (cost is not None) else amount * price
-        upperCaseType = type.upper()
+        upper_case_type = type.upper()
         request = {
             'symbol': market['id'],
             'side': self.capitalize(side),
-            'orderType': upperCaseType,  # limit, market or limit_maker
+            'orderType': upper_case_type,  # limit, market or limit_maker
             'timeInForce': 'GTC',  # FOK, IOC
-            'orderQty': self.amount_to_precision(symbol, amount),
             # 'orderLinkId': 'string',  # unique client order id, max 36 characters
         }
-        if (upperCaseType == 'LIMIT') or (upperCaseType == 'LIMIT_MAKER'):
+        if (type == 'market') and (side == 'buy'):
+            if self.options['createMarketBuyOrderRequiresPrice']:
+                if price is None:
+                    raise InvalidOrder(self.id + " market buy order requires price argument to calculate cost(total amount of quote currency to spend for buying, amount * price). To switch off self warning exception and specify cost in the amount argument, set .options['createMarketBuyOrderRequiresPrice'] = False. Make sure you know what you're doing.")
+                else:
+                    # for market buy it requires the amount of quote currency to spend
+                    request['orderQty'] = self.cost_to_precision(symbol, float(amount) * float(price))
+            else:
+                request['orderQty'] = self.cost_to_precision(symbol, amount)
+        else:
+            request['orderQty'] = self.amount_to_precision(symbol, amount)
+
+        if (upper_case_type == 'LIMIT') or (upper_case_type == 'LIMIT_MAKER'):
             if price is None:
                 raise InvalidOrder(self.id + ' createOrder requires a price argument for a ' + type + ' order')
             request['orderPrice'] = float(self.price_to_precision(symbol, price))
-        isPostOnly = self.is_post_only(upperCaseType == 'MARKET', type == 'LIMIT_MAKER', params)
-        if isPostOnly:
+        is_post_only = self.is_post_only(upper_case_type == 'MARKET', type == 'LIMIT_MAKER', params)
+        if is_post_only:
             request['orderType'] = 'LIMIT_MAKER'
-        clientOrderId = self.safe_string_2(params, 'clientOrderId', 'orderLinkId')
-        if clientOrderId is not None:
-            request['orderLinkId'] = clientOrderId
-        params = self.omit(params, ['clientOrderId', 'orderLinkId', 'postOnly'])
+        client_order_id = self.safe_string(params, 'order_link_id')
+        if client_order_id is not None:
+            request['orderLinkId'] = client_order_id
+        params = self.omit(params, ['order_link_id', 'post_only'])
```

Comment:
no post_only?

Reply:
it's being used inside the following function

is_post_only = self.is_post_only(upper_case_type == 'MARKET', type == 'LIMIT_MAKER', params)